crates/router/src/core/errors/api_error_response.rs

```diff
@@ -182,6 +182,8 @@ pub enum ApiErrorResponse {
     MissingFilePurpose,
     #[error(error_type = ErrorType::InvalidRequestError, code = "HE_04", message = "File content type not found / valid")]
     MissingFileContentType,
+    #[error(error_type = ErrorType::RouterError, code = "RE_01", message = "Failed to remove redis cache key")]
+    RemoveRedisKeyFailure,
```

Comment:
@dracarys18 @Narayanbhat166 Please confirm if we would need a new error to be introduced for this scenario, or if we would prefer an `InternalServerError` being raised instead.

Reply:
we can use InternalServerError in its place. This function/route does only one thing, that is to remove the key from redis, there are not a lot of moving parts. This error tells that `RedisKeyRemovalFailed`, in order to know why it failed we anyway have to look into the logs and debug. So This error will not add much value over using `InternalServerError`.